### PR TITLE
Remove need for port parameter

### PR DESCRIPTION
**Pull request by rtluckie**
- automatically assign port
- remove need to specify port when running multiple KDKs
- update README

### CODE REVIEW

File: cmd/kdk/init.go [hunk 1]
```diff
@@ -34,8 +34,9 @@ var initCmd = &cobra.Command{
 }
 
 func init() {
+	CurrentKdkEnvConfig.ConfigFile.AppConfig.Port = kdk.Port
+
 	initCmd.Flags().StringVarP(&CurrentKdkEnvConfig.ConfigFile.AppConfig.Name, "name", "n", "kdk", "KDK Name")
-	initCmd.Flags().StringVarP(&CurrentKdkEnvConfig.ConfigFile.AppConfig.Port, "port", "p", "2022", "KDK Port")
```
Comment (josdotso) [inline cmd/kdk/init.go]: Shouldn't it still be possible to override with flag?  Default could be read back from conf, yeah?

File: pkg/kdk/config.go [hunk 2]
```diff
@@ -16,16 +16,21 @@ package kdk
 
 import (
 	"context"
+	"github.com/cisco-sso/kdk/pkg/utils"
```
Comment (josdotso) [inline pkg/kdk/config.go]: Should: Move the github dep down to the non-core libs list which starts at line 25.

File: pkg/kdk/ssh.go [hunk 3]
```diff
@@ -21,11 +21,44 @@ import (
...
 	"github.com/Sirupsen/logrus"
 	"github.com/codeskyblue/go-sh"
+	"github.com/docker/docker/api/types"
 )
 
-func Ssh(cfg KdkEnvConfig, logger logrus.Entry) error {
+func Ssh(cfg KdkEnvConfig, logger logrus.Entry) {
 	logger.Info("Connecting to KDK container")
 
+	// Pull KDK image
```
Comment (josdotso) [inline pkg/kdk/ssh.go]: Does this belong in the ssh.go file / ssh function?  Seems like an init thing (following example of vagrant)

Comment (rtluckie) [reply pkg/kdk/ssh.go]: intent was to minimize the number of commands required to get the KDK running.
so  `kdk init && kdk pull && kdk up && kdk ssh` -> `kdk init && kdk ssh`

File: pkg/kdk/ssh.go [hunk 4]
```diff
@@ -21,11 +21,44 @@ import (
...
+		for _, name := range container.Names {
+			if name == "/"+cfg.ConfigFile.AppConfig.Name {
+				kdkRunning = true
+				break
+			}
+		}
+	}
+
+	// if KDK container is not running, start it and provision KDK user
+	if !kdkRunning {
```
Comment (dcwangmit01) [inline pkg/kdk/ssh.go]: This bug will most certainly happen here:
https://github.com/cisco-sso/kdk/issues/50

**Comment by dcwangmit01** [comment]
Also, in a followup PR please select the port from a specific range.  Not random.

File: pkg/kdk/ssh.go (hunk 4 above)
Comment (rtluckie) [reply pkg/kdk/ssh.go]: I'll work on fixing #50 in follow up PR.

File: pkg/kdk/config.go (hunk 2 above)
Comment (rtluckie) [reply pkg/kdk/config.go]: good catch.
fixed.

**Pull request merged**